addons/web/static/src/core/popover/popover_service.js

```diff
@@ -125,6 +139,36 @@ export const popoverService = {
             },
         };
     },
+    specializeForComponent(component, service) {
+        const keys = new Set();
+        onWillUnmount(function () {
+            for (const key of keys) {
+                service.remove(key);
+            }
+            keys.clear();
+        });
+        return Object.assign(Object.create(service), {
+            add(params) {
+                const newParams = Object.create(params);
+                newParams.onClose = function (key) {
+                    if (!params.keepOnClose) {
+                        // manager will delete the popover if keepOnClose is falsy
+                        keys.delete(key);
+                    }
+                    if (params.onClose && component.__owl__.status !== 5 /* DESTROYED */) {
+                        params.onClose(key);
+                    }
+                };
+                const key = service.add(newParams);
+                keys.add(key);
```

Comment:
would it not be slightly more natural to do 
```js
if (!params.keepOnClose) {
  keys.add(key);
}
```
and to remove the if statement above?